src/documents/actor/character.ts

```diff
@@ -646,14 +677,15 @@ export class NimbleCharacter extends NimbleBaseActor {
 		});
 
 		// Add selected subclass if available
-		const subclass = dialogData.selectedSubclass as NimbleSubclassItem;
+		const subclass = typedDialogData.selectedSubclass;
 
 		if (subclass) {
 			if (subclass.system.parentClass === characterClass.identifier) {
 				// Check if this subclass is actually for this class
 				// Create a copy of the subclass for the character
 				const subclassData = subclass.toObject();
-				subclassData._stats.compendiumSource = subclass.uuid;
+				(subclassData as { _stats: { compendiumSource?: string } })._stats.compendiumSource =
+					subclass.uuid;
```

Comment:
I don't like the concept of assigning and then having to type cast for a specific case. Instead we should have assigning with type guard and tehn it is the right type.